lib/kernel/src/code.erl

```diff
@@ -187,10 +187,10 @@ ensure_loaded(Mod) when is_atom(Mod) ->
     case erlang:module_loaded(Mod) of
         true -> {module, Mod};
         false ->
-            case get_object_code(Mod) of
-                error -> call({sync_ensure_on_load, Mod});
-                {Mod,Binary,File} ->
-                    load_module(Mod, File, Binary, false, true)
+            case call({get_object_code_for_loading, Mod}) of
+                {module, Mod} -> {module, Mod};
+                {error, What} -> {error, What};
+                {Mod,Binary,File,Ref} -> load_module(Mod, File, Binary, false, Ref)
```

Comment:
I think we can still keep the `get_object_code` in this process and not in the code server, something like:

```suggestion
            case call({lock_for_loading, Mod}) of
                {module, Mod} -> {module, Mod};
                {error, What} -> {error, What};
                {go,Ref} ->
                  case get_object_code(Mod) of
                    :error -> call({no_module_to_load, Mod, Ref});
                    {Mod, Binary, File} -> load_module(Mod, File, Binary, false, Ref)
                  end
```

This should be the best of both worlds because we keep the work on the server minimal while avoiding concurrent work?

Reply:
Ignore me, the get object code needs to happen on the server. :)